### PR TITLE
FPGA update to support I2C clock stretching and ack-polling

### PR DESCRIPTION
**Pull request by Aaron-Hartwig**
Impacts the Front IO FPGAs which have I2C interfaces to transceiver modules, some of which may clock stretch. Also impacts the Mainboard FPGA as it has an I2C interface to the Tofino 2. In addition to clock stretching, the core will now also perform acknowledge-polling after all writes.

This can land after https://github.com/oxidecomputer/transceiver-control/pull/278 and https://github.com/oxidecomputer/quartz/pull/210.

### CODE REVIEW

**Review by arjenroodselaar — APPROVED** [review]
Thank you for the work here, looking forward to this getting deployed!

**Comment by Aaron-Hartwig** [comment]
@mkeeter throwing you on here as I'm touching the transceiver thermal loop's port-disabling mechanism to add in another potential failure. The vast majority of the diff is general FPGA update noise.

File: drv/sidecar-front-io/src/transceivers.rs [hunk 1]
```diff
@@ -687,6 +688,9 @@ impl PortI2CStatus {
     pub fn new(status: u8) -> Self {
         // Use QSFP::PORT0 for constants, since they're all identical
         Self {
+            stretching_seen: (status
+                & Reg::QSFP::PORT0_STATUS::STRETCHING_SEEN)
+                != 0,
```
Comment (mkeeter) [inline drv/sidecar-front-io/src/transceivers.rs:693]: wow `rustfmt` is mangling this expression

Very optional, but could we get less terrible formatting if we imported `reg::QSFP::PORT0_STATUS`, then did

```
(status & PORT0_STATUS::STRETCHING_SEEN) != 0
```

File: drv/transceivers-server/src/main.rs [hunk 2]
```diff
@@ -84,16 +84,17 @@ counted_ringbuf!(Trace, 16, Trace::None);
 
 ////////////////////////////////////////////////////////////////////////////////
 
-/// After seeing this many NACKs, we disable the port by policy.
+/// After seeing this many NACKs or timeouts, we disable the port by policy.
 ///
 /// This should be **very rare**: it requires a transceiver to correctly report
 /// its type (SFF vs CMIS) over I2C when it's first plugged in, but then begin
-/// NACKing while still physically present (according to the `modprsl` pin).
+/// NACKing or timing out while still physically present
```
Comment (mkeeter) [inline drv/transceivers-server/src/main.rs:91]: nit: wordwrapping

**Review by mkeeter — APPROVED** [review]
Hubris changes look pretty trivial, ship it

**Pull request merged**